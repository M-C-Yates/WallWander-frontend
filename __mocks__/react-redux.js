jest.mock('react-redux', () => {
	return {
		useDispatch() {
			return { dispatch: jest.fn() };
		},
		useSelector() {
			return {
				id: '12345',
				email: 'foo@bar.com',
				name: 'foobar',
				token: '123abc',
				nsfw: false
			};
		}
	};
});
