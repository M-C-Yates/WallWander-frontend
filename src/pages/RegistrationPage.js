import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

import { startRegistration } from '../actions/auth';
import { UserForm } from '../components/UserForm';

const RegistrationPage = (props) => {
	const [email, setEmail] = useState('');
	const [password, setPassword] = useState('');

	const dispatch = useDispatch();

	const register = (e) => {
		e.preventDefault();
		const user = { email, password };
		dispatch(startRegistration(user));

		props.navigate('/');
	};

	return (
		<div className="container">
			<UserForm
				setEmail={setEmail}
				setPassword={setPassword}
				submit={register}
				button="Register"
			/>
		</div>
	);
};

export default RegistrationPage;
