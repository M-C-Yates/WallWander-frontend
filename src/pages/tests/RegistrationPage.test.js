import React from 'react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import { render, cleanup } from '@testing-library/react';
import { useDispatch } from 'react-redux';

import RegistrationPage from '../RegistrationPage';

afterEach(cleanup);

describe('RegistrationPage', () => {
	it('should match snapshot', () => {
		const { asFragment } = render(<RegistrationPage />);
		expect(asFragment).toMatchSnapshot();
	});
});
