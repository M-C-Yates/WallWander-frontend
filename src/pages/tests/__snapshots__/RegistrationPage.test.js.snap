// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage should match snapshot 1`] = `[Function]`;
